pkg/agent/multicast/kernel_source.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package multicast
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+// We should consider the changes in vifctl struct after kernel versiopn 5.9
+func parseKernelVersion() (*kernelVersion, error) {
+	var uname syscall.Utsname
+	if err := syscall.Uname(&uname); err != nil {
+		return nil, fmt.Errorf("running uname error: %v", err)
+	}
+	kernel_version_full := arrayToString(uname.Release)
+	kernel_version_numbers := strings.Split(kernel_version_full, "-")
+	kernel_version_array := strings.Split(kernel_version_numbers[0], ".")
+	major, err := strconv.Atoi(kernel_version_array[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse kernel version %s", kernel_version_full)
+	}
+	minor, err := strconv.Atoi(kernel_version_array[1])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse kernel version %s", kernel_version_full)
+	}
+	patch, err := strconv.Atoi(kernel_version_array[2])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse kernel version %s", kernel_version_full)
+	}
+	return &kernelVersion{
+		major: major,
+		minor: minor,
+		patch: patch,
+	}, nil
+}
+
+func arrayToString(x [65]int8) string {
+	var buf [65]byte
+	for i, b := range x {
+		buf[i] = byte(b)
+	}
+	str := string(buf[:])
+	if i := strings.Index(str, "\x00"); i != -1 {
+		str = str[:i]
+	}
+	return str
+}
+
+type kernelVersion struct {
+	major int
+	minor int
+	patch int
+}
+
+type parsedIgmpMsg struct {
+	imMsgtype uint8
```

Comment:
imMsgType

Question - all other fields in this struct are public (capitalized), but this one is not. Is it intentional?

Reply:
this field is removed